### ISSUE TITLE
Built-in linking takes precedence over the values set in the setup function

### ISSUE THREAD

**Issue by olimorris** [olimorris/onedarkpro.nvim#196]
### Discussed in https://github.com/olimorris/onedarkpro.nvim/discussions/195

<div type='discussions-op-text'>

<sup>Originally posted by **anthony-S93** June 20, 2023</sup>
@olimorris 

I did an experiment by setting: 
```lua
highlights = {
    ["@keyword"] = { fg = "${purple}", style = "bold" },
}
-- Set "@keyword" while leaving the "Keyword" highlight group in its default value 
```

The `:hi @keyword` command showed something surprising, at least to me.

![2023-06-20-21-14-01](https://github.com/olimorris/onedarkpro.nvim/assets/69449791/f45cc1c2-36fc-4056-9224-2baf10059fd4)

I expected the `@keyword` highlight group to be overridden by the setting in the setup function, but that appears to not be the case. The highlight group was still linked to the `Keyword` highlight group (which I assume was done internally in the plugin). Not even un-setting the highlight group `@keyword = {} ` has any effect, which seems a bit counterintuitive. 

The settings in the setup function will only apply if the group isn't internally (hard coded?) linked to other groups. E.g: `@conditional` 
```lua
highlights = {
    ["@conditional"] = {fg = "${purple}", style = "bold"}, -- this one works
}
```
![2023-06-20-21-22-56](https://github.com/olimorris/onedarkpro.nvim/assets/69449791/b726a1ac-aefc-4095-ae6f-abffdd5c9d65)
</div>

**Comment by olimorris** [comment]
This was introduced in a280557fa22b7aa7feff31617e31baca3d6c0392.

Now, by default, a users highlights extend and don't overwrite existing highlight groups. As can be seen below, this will yield to strange outputs:

![2023-06-20-21-14-01](https://github.com/olimorris/onedarkpro.nvim/assets/69449791/f45cc1c2-36fc-4056-9224-2baf10059fd4).

**Comment by olimorris** [comment]
This has been fixed in 0b53cb1a89f502d10934a4afc8a276a797330f17.

Now, to extend a highlight:

```lua
require("onedarkpro").setup({
  highlights = {
    Comment = { underline = true, extend = true }
  }
})
```

cc: @maxmx03
